integration/src/test/scala/zio/jdbc/DuckDbSpec.scala

```diff
@@ -0,0 +1,78 @@
+package zio.jdbc
+
+import org.duckdb.DuckDBConnection
+import org.testcontainers.jdbc.ConnectionWrapper
+import zio._
+import zio.schema.{ DeriveSchema, Schema }
+import zio.test.TestAspect.{ repeats, sequential, shrinks, withLiveClock }
+import zio.test.{ Gen, Spec, TestEnvironment, ZIOSpec, assertTrue, check }
+
+import java.sql.{ Array => _, _ }
+import java.time.{ OffsetDateTime, ZoneOffset }
+import java.time.temporal.ChronoUnit
+import java.util.Properties
+
+object DuckDbSpec extends ZIOSpec[ZConnection] {
+
+  override def bootstrap: ZLayer[Any, Any, ZConnection] =
+    ZLayer.scoped(
+      for {
+        connection  <- ZIO.acquireRelease(
+                         ZIO.succeedBlocking {
+                           val props  = new Properties()
+                           val duckDb = DriverManager
+                             .getConnection(s"jdbc:duckdb:", props)
+                             .asInstanceOf[DuckDBConnection]
+                           WorkaroundDuckDBConnection(duckDb)
+                         }
+                       )(c => ZIO.succeed(c.close()))
+        zConnection <- ZConnection.make(connection)
+      } yield zConnection
+    )
+
+  case class OffsetDateTimeRow(value: OffsetDateTime)
```

Comment:
When using query[OffsetDateTime] directly things were working already, the code path I changed in JdbcDecoder is only hit when using the JdbcDecoder.fromSchema

Reply:
Without the change, the case class would have had to have an Instant field instead of an OffsetDateTime field. 
The data would be returned in the default system timezone, but interpreted as a timestamp in UTC. Where I live this results in a 2 hour offset currently between writing and reading an OffsetDateTimeRow